hack/generator/pkg/astmodel/struct_type.go

```diff
@@ -80,18 +80,16 @@ func (structType *StructType) RequiredImports() []*PackageReference {
 	return result
 }
 
-// References this type has to the given type
-func (structType *StructType) References(d *TypeName) bool {
-
+// References returns the combination of all the types the fields refer to.
+func (structType *StructType) References() TypeNameSet {
+	var results TypeNameSet
 	for _, field := range structType.fields {
-		if field.FieldType().References(d) {
-			return true
+		for ref := range field.FieldType().References() {
+			results = results.Add(ref)
 		}
 	}
-
-	// For now, not considering functions in references on purpose
-
-	return false
+	// Not collecting types from functions deliberately.
```

Comment:
Ok for now, but when we start generation conversion functions, I think we'll need to do this.